Client/Frontend/AuthenticationManager/AuthenticationSettingsViewController.swift

```diff
@@ -124,7 +124,19 @@ class TouchIDSetting: Setting {
     private weak var navigationController: UINavigationController?
     private weak var switchControl: UISwitch?
 
-    init(title: NSAttributedString?, navigationController: UINavigationController? = nil, delegate: SettingsDelegate? = nil, enabled: Bool? = nil) {
+    private var touchIDSuccess: (() -> Void)? = nil
+    private var touchIDFallback: (() -> Void)? = nil
+
+    init(
+        title: NSAttributedString?,
+        navigationController: UINavigationController? = nil,
+        delegate: SettingsDelegate? = nil,
+        enabled: Bool? = nil,
+        touchIDSuccess: (() -> Void)? = nil,
+        touchIDFallback: (() -> Void)? = nil)
```

Comment:
Holding callbacks like this without `weak` is a red flag for retain cycles. Sure enough, I just tested this locally and can verify that this does cause a retain loop -- try adding `print("deinit") to the `AuthenticationSettingsViewController`'s deinitializer.